### PR TITLE
V4 - Additional improvements

### PR DESCRIPTION
**Pull request by nagi1**
- [x] Add docblock for some undocumented public functions.
- [x] Add ability to get batch uuid in `LogBatch::withinBatch()`.
- [x] Add ability to compare date intervals casts.

### CODE REVIEW

File: src/Traits/LogsActivity.php [hunk 1]
```diff
@@ -310,6 +313,14 @@ function ($new, $old) {
                         return $new === $old ? 0 : 1;
                     }
 
+                    // Handels Date intervels comparsons since php cannot use spaceship
```
Comment (Gummibeer) [inline src/Traits/LogsActivity.php:316]: Could you add 3 testcases for this new logic? The code was out of my head and in no way tested.

```
null => DateInterval
DateInterval#1 => DateInterval#2
DateInterval => null
```

**Review by Gummibeer — CHANGES_REQUESTED** [review]

**Comment by Gummibeer** [comment]
@nagi1 can you take care of that?

**Comment by nagi1** [comment]
Will do, Can you give me example use case to work with?

**Comment by Gummibeer** [comment]
A new attribute with, I think, custom object cast to DateInterval instance.
And a testcase covering at least the three listed scenarios.

`$model->interval = CarbonInterval::minute()` for example.

**Comment by nagi1** [comment]
@Gummibeer You may merge this now :)

File: src/Traits/LogsActivity.php [hunk 2]
```diff
@@ -316,9 +316,9 @@ function ($new, $old) {
                     // Handels Date intervels comparsons since php cannot use spaceship
                     // Operator to compare them and will throw ErrorException.
                     if ($old instanceof DateInterval) {
-                        return CarbonInterval::make($old)->equalTo($new);
+                        return intval(! CarbonInterval::make($old)->equalTo($new));
```
Comment (Gummibeer) [inline src/Traits/LogsActivity.php:319]: Could you use something like `CarbonInterval::make($old)->equalTo($new) ? 0 : 1` - the current one is correct but not really readable.

**Review by Gummibeer — CHANGES_REQUESTED** [review]

**Pull request merged**